sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -909,19 +909,24 @@ object ScalaJSPluginInternal {
         (scalaJSModuleInitializers in (This, Zero, This)).value,
 
       scalaJSModuleInitializers ++= {
+        val mainClasses = scalaJSDiscoveredMainClasses.value
         if (scalaJSUseMainModuleInitializer.value) {
           Seq(scalaJSMainModuleInitializer.value.getOrElse {
-            throw new MessageOnlyException(
-                "No main module initializer was specified (possibly because " +
-                "no or multiple main classes were found), but " +
+            if (mainClasses.isEmpty)
+              throw new MessageOnlyException(
+                "No main module initializer was specified, but " +
                 "scalaJSUseMainModuleInitializer was set to true. " +
                 "You can explicitly specify it either with " +
                 "`mainClass := Some(...)` or with " +
                 "`scalaJSMainModuleInitializer := Some(...)`")
+            else
+              throw new MessageOnlyException(
+                s"Multiple main classes (${mainClasses.keys}) were found. " +
+                "You can explicitly specify the one you want with " +
+                "`mainClass := Some(...)` or with " +
+                "`scalaJSMainModuleInitializer := Some(...)`")
           })
-        } else {
-          Seq.empty
-        }
+        } else Seq.empty
```

Comment:
The else should be formatted as before: with `{}` and with `Seq.empty` on a separate line, according to our coding style.